app/controllers/api/v1/customers_controller.rb

```diff
@@ -10,15 +10,29 @@ class CustomersController < Api::V1::BaseController
       skip_authorization_check only: :index
 
       before_action :authorize_action, only: [:show, :update, :destroy]
+      # Query parameters
+      before_action :set_display_customer_balance, only: [:index] # Always show balance for show
 
       def index
         @pagy, customers = pagy(search_customers, pagy_options)
-
-        render json: Api::V1::CustomerSerializer.new(customers, pagination_options)
+        render json: Api::V1::CustomerSerializer.new(
+          customers,
+          pagination_options.merge({
+                                     params: {
+                                       display_customer_balance: @display_customer_balance
+                                     }
+                                   })
+        )
       end
 
       def show
-        render json: Api::V1::CustomerSerializer.new(customer, include_options)
+        render json: Api::V1::CustomerSerializer.new(
+          CustomersWithBalance.new(customers: customer).query.first,
+          {
+            include: [params.fetch(:include, [])].flatten.map(&:to_s),
+            params: { display_customer_balance: true },
```

Comment:
This API is following JSON:API. There is a standardised way to include attributes like this with the `fields` option. We should probably use that. Have a look at https://jsonapi.org/examples/.

Using the `fields` option hides all other fields but I think that's acceptable for our use case. The API call can mention all desired fields. Is that okay @lin-d-hop?

Reply:
Yes if that is standard practice then I say we go with that :+1: